File: python/java_to_python/05-pgmAbstraccion/src/co/edu/campusucc/poo/MainGrandPrix.py
from Vehiculo import Vehiculo
from TipoCombustion import TipoCombustion
from TipoElectrico import TipoElectrico


def main():
    vehiculo = TipoCombustion()
    vehiculo.setMarca("Toyota")
    vehiculo.setCarroseria("Truck")
    print(vehiculo)

    vehiculo.startingEngine()

    vehiculo.speedUp()
    vehiculo.breaking()
    vehiculo.currentSpeed()
    vehiculo.parking()
    vehiculo.stopingEngine()

    vehiculo = TipoElectrico()
    vehiculo.setMarca("Tesla")
    vehiculo.setCarroseria("Sedan")
    print(vehiculo)

    vehiculo.startingEngine()
    vehiculo.speedUp()
    vehiculo.speedUp()
    vehiculo.currentSpeed()
    vehiculo.breaking()
    vehiculo.speedUp()
    vehiculo.breaking()
    vehiculo.currentSpeed()
    vehiculo.parking()
    vehiculo.stopingEngine()


if __name__ == '__main__':
    main()

File: ruby/pgms_ruby/05-pgmAbstraccion/MainGrandPrix.rb
require_relative 'Vehiculo'
require_relative 'TipoCombustion'
require_relative 'TipoElectrico'

def main
  vehiculo = TipoCombustion.new
  vehiculo.setMarca("Toyota")
  vehiculo.setCarroseria("Truck")
  puts vehiculo

  vehiculo.startingEngine

  vehiculo.speedUp
  vehiculo.breaking
  vehiculo.currentSpeed
  vehiculo.parking
  vehiculo.stopingEngine

  vehiculo = TipoElectrico.new
  vehiculo.setMarca("Tesla")
  vehiculo.setCarroseria("Sedan")
  puts vehiculo

  vehiculo.startingEngine
  vehiculo.speedUp
  vehiculo.speedUp
  vehiculo.currentSpeed
  vehiculo.breaking
  vehiculo.speedUp
  vehiculo.breaking
  vehiculo.currentSpeed
  vehiculo.parking
  vehiculo.stopingEngine
end

if __FILE__ == $0
  main
end